admin/database/postgres/migrations/0007.sql

```diff
@@ -0,0 +1,102 @@
+CREATE TABLE org_roles (
+    id UUID DEFAULT uuid_generate_v4() PRIMARY KEY,
+    name TEXT NOT NULL,
+    read_org boolean,
+    manage_org boolean,
+    read_projects boolean,
+    create_projects boolean,
+    manage_projects boolean,
+    read_org_members boolean,
+    manage_org_members boolean
+);
+
+CREATE UNIQUE INDEX org_role_name_idx ON org_roles (name);
+
+CREATE TABLE project_roles (
+    id UUID DEFAULT uuid_generate_v4() PRIMARY KEY,
+    name TEXT NOT NULL,
+    read_project boolean,
+    manage_project boolean,
+    read_prod_branch boolean,
+    manage_prod_branch boolean,
+    read_dev_branches boolean,
+    manage_dev_branches boolean,
+    read_project_members boolean,
+    manage_project_members boolean
+);
+
+CREATE UNIQUE INDEX project_role_name_idx ON project_roles (name);
+
+INSERT INTO org_roles (id, name, read_org, manage_org, read_projects, create_projects, manage_projects, read_org_members, manage_org_members)
+VALUES
+    ('12345678-0000-0000-0000-000000000001', 'admin', true, true, true, true, true, true, true),
+    ('12345678-0000-0000-0000-000000000002', 'collaborator', true, false, true, false, false, true, false),
+    ('12345678-0000-0000-0000-000000000003', 'reader', true, false, true, false, false, true, false);
+
+INSERT INTO project_roles (id, name, read_project, manage_project, read_prod_branch, manage_prod_branch, read_dev_branches, manage_dev_branches, read_project_members, manage_project_members)
+VALUES
+    ('12345678-0000-0000-0000-000000000001', 'admin', true, true, true, true, true, true, true, true),
+    ('12345678-0000-0000-0000-000000000002', 'collaborator', true, false, true, false, true, true, true, false),
+    ('12345678-0000-0000-0000-000000000003', 'reader', true, false, true, false, true, false, false, false);
+
+CREATE TABLE users_orgs_roles (
+    user_id UUID NOT NULL REFERENCES users (id) ON DELETE CASCADE,
+    org_id UUID NOT NULL REFERENCES organizations (id) ON DELETE CASCADE,
+    org_role_id UUID NOT NULL REFERENCES org_roles (id) ON DELETE CASCADE,
+    PRIMARY KEY (user_id, org_id)
+);
+
+CREATE TABLE users_projects_roles (
+    user_id UUID NOT NULL REFERENCES users (id) ON DELETE CASCADE,
+    project_id UUID NOT NULL REFERENCES projects (id) ON DELETE CASCADE,
+    project_role_id UUID NOT NULL REFERENCES project_roles (id) ON DELETE CASCADE,
+    PRIMARY KEY (user_id, project_id)
+);
+
+-- user group related table operations
+CREATE TABLE usergroups (
+    id UUID DEFAULT uuid_generate_v4() PRIMARY KEY,
+    org_id UUID NOT NULL REFERENCES organizations (id) ON DELETE CASCADE,
+    display_name TEXT NOT NULL,
+    description TEXT
+);
+
+-- create all users group for existing organizations
+INSERT INTO usergroups (org_id, display_name) SELECT id, 'all_users' FROM organizations;
```

Comment:
If usergroup names will be shown to users, should this be something like `'All organization members'` instead?

Reply:
Got it. I was picturing that we would always use the hard-coded role UUIDs, and just do a `switch` from hard-coded names to those UUIDs on the CLI side.

I'm also up for the CLI-friendly `name` + user-friendly `description` approach if you prefer that. In that case, we can get rid of the hard-coded UUIDs, right?

Reply:
Nah, let's keep it simple for now :)